sscanss/core/geometry/primitive.py

```diff
@@ -6,6 +6,73 @@
 from ..math.transform import rotation_btw_vectors
 
 
+def create_cone(radius=1.0, height=1.0, slices=64, stacks=1, closed=True):
+    """Generates the vertices, normals, and indices for a cone mesh
+
+    :param radius: cone base radius
+    :type radius: float
+    :param height: cone height
+    :type height: float
+    :param slices: number of radial segments to use
+    :type slices: int
+    :param stacks: number of height segments to use
+    :type stacks: int
+    :param closed: indicates if mesh should be closed at the bottom
+    :type closed: bool
+    :return: The vertices, normals and index array of the mesh
+    :rtype: Mesh
+    """
+
+    theta = np.linspace(0, 2 * np.pi, slices + 1)[:-1]
+
+    # Add the cone
+    r = np.repeat(np.linspace(1.0, 0.0, stacks + 1), slices) * radius
+
+    x = np.tile(np.sin(theta), stacks + 1) * r
+
+    y = np.tile(np.cos(theta), stacks + 1) * r
+
+    z_div = np.linspace(0.0, 1.0, stacks + 1)
+    z = np.repeat(z_div * height, slices)
```

Comment:
Sorry I forgot to mention could you make it so the origin is at the centre of the cone, i.e. shift the z by `-height/2`